src/overlays/actors/ovl_Dm_Hina/z_dm_hina.c

```diff
@@ -33,24 +32,156 @@ ActorInit Dm_Hina_InitVars = {
+    s32 var_a0;
+
+    switch (this->unk17E) {
+        case 0:
+            this->unk17F = 0;
+            this->unk14C = 0.0f;
+            break;
+        case 1:
+            Math_SmoothStepToF(&this->unk14C, 1.0f, 0.4f, 0.05f, 0.001f);
+            this->unk17F = this->unk14C * 255.0f;
+            this->unk150 = Math_SinS(play->state.frames * 0x1F40);
+            for (var_a0 = 0; var_a0 < 3; var_a0++) {
+                play->envCtx.lightSettings.ambientColor[var_a0] = play->envCtx.lightSettings.fogColor[var_a0] =
+                    play->envCtx.lightSettings.diffuseColor1[var_a0] = -255.0f * this->unk14C;
+            }
+            play->envCtx.lightSettings.fogNear = -500.0f * this->unk14C;
+            if (play->envCtx.lightSettings.fogNear < -300) {
+                play->roomCtx.curRoom.segment = NULL;
+            }
+            break;
+    }
+}
+
+void DmHina_Update(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    this->actionFunc(this, play);
+    func_80A1F75C(this, play);
+}
+
+extern UNK_TYPE D_04023348;
+extern UNK_TYPE D_04023428;
+
+void func_80A1F9AC(DmHina* this, PlayState* play) {
+    GraphicsContext* gfxCtx = play->state.gfxCtx;
+
+    if (this->unk17E != 0) {
+        OPEN_DISPS(gfxCtx);
+
+        Gfx_SetupDL25_Xlu(play->state.gfxCtx);
+
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, (u8)(this->unk14C * 100.0f) + 0x9B, this->unk17F);
+        gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, (u8)(this->unk150 * 100.0f) + 0x32, 0);
+        gSPDisplayList(POLY_XLU_DISP++, &D_04023348);
+
+        Matrix_Translate(this->actor.world.pos.x,
+                         this->actor.world.pos.y + (this->unk154 * this->unk_15C) + (40.0f * this->unk_15C),
+                         this->actor.world.pos.z, MTXMODE_NEW);
+        Matrix_ReplaceRotation(&play->billboardMtxF);
+        Matrix_Scale(this->unk14C * 20.0f, this->unk14C * 20.0f, this->unk14C * 20.0f, MTXMODE_APPLY);
+        Matrix_RotateZF(Rand_ZeroFloat(6.2831855f), MTXMODE_APPLY);
+
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, &D_04023428);
+
+        CLOSE_DISPS(gfxCtx);
+    }
+}
+
+void DmHina_Draw(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    if ((u8)this->unk180 != 0) {
+        Matrix_Translate(this->actor.world.pos.x,
+                         this->actor.world.pos.y + (this->unk154 * this->unk_15C) + (40.0f * this->unk_15C),
+                         this->actor.world.pos.z, MTXMODE_NEW);
+        Matrix_RotateZYX(0, (s16)(play->gameplayFrames * 0x3E8), 0, MTXMODE_APPLY);
+        Matrix_Scale((this->unk148 * (1.0f - this->unk14C) * this->unk_15C),
+                     (this->unk148 * (1.0f - this->unk14C) * this->unk_15C),
+                     (this->unk148 * (1.0f - this->unk14C) * this->unk_15C), MTXMODE_APPLY);
+        switch (this->actor.params) {
+            case 0:
+                GetItem_Draw(play, 0x5C);
```

Comment:
```suggestion
                GetItem_Draw(play, GID_REMAINS_ODOLWA);
```
Again looking up usages for the function show the proper enums to use. Please fill out the ones as well.